Model/AdyenInitiateTerminalApi.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2018 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+
+namespace Adyen\Payment\Model;
+
+
+use Adyen\Payment\Api\AdyenInitiateTerminalApiInterface;
+use Adyen\Payment\Model\Ui\AdyenPosCloudConfigProvider;
+use Adyen\Util\Util;
+use Magento\Payment\Gateway\Http\ClientInterface;
+
+class AdyenInitiateTerminalApi implements AdyenInitiateTerminalApiInterface
+{
+
+    private $_encryptor;
+    private $_adyenHelper;
+    private $_adyenLogger;
+    private $_recurringType;
+    private $_appState;
+
+    protected $_quoteRepository;
+
+    /**
+     * AdyenInitiateTerminalApi constructor.
+     * @param \Magento\Framework\Model\Context $context
+     * @param \Magento\Framework\Encryption\EncryptorInterface $encryptor
+     * @param \Adyen\Payment\Helper\Data $adyenHelper
+     * @param \Adyen\Payment\Logger\AdyenLogger $adyenLogger
+     * @param RecurringType $recurringType
+     * @param array $data
+     */
+    public function __construct(
+        \Magento\Framework\Model\Context $context,
+        \Magento\Framework\Encryption\EncryptorInterface $encryptor,
+        \Adyen\Payment\Helper\Data $adyenHelper,
+        \Adyen\Payment\Logger\AdyenLogger $adyenLogger,
+        \Adyen\Payment\Model\RecurringType $recurringType,
+        \Magento\Quote\Api\CartRepositoryInterface $quoteRepository,
+        array $data = []
+    ) {
+        $this->_encryptor = $encryptor;
+        $this->_adyenHelper = $adyenHelper;
+        $this->_adyenLogger = $adyenLogger;
+        $this->_recurringType = $recurringType;
+        $this->_appState = $context->getAppState();
+        $this->_quoteRepository = $quoteRepository;
+
+        // initialize client
+        $apiKey = $this->_adyenHelper->getApiKey();
+        $client = new \Adyen\Client();
```

Comment:
This client constructor is duplicated in all transaction objects. I believe is good time to move it on helper or client factory.

Reply:
created a separate issue for this